### PR TITLE
Accuracy Function For Speeding Up LSTM+CTC Convergence

### PR DESCRIPTION
**Pull request by catalystfrank**
When processing Sequence Tagging problems, an accuracy func which is too strict is not convergence-friendly. This accuracy function is from the solution of a basic leetcode problem LCS (Largest Common Subsequence). Original All-Correct-Or-Nothing accuracy function may takes 7x times (or more) long to achieve certain accuracy.

### CODE REVIEW

**Comment by catalystfrank** [comment]
It seems that this PR is irrelevant to Travis-CI build failure on R_test.
Please check.

**Comment by piiswrong** [comment]
@xlvector How about make this the default?

**Comment by xlvector** [comment]
Yes, LCS is better. @catalystfrank what about training speed. Sometimes, I find eval function is cost of time. Can you tell me the samples/second before/after using this evaluate metric?

**Comment by catalystfrank** [comment]
At the end of 1st epoch, log shows

2017-02-04 08:49:46,279 Epoch[0] Batch [9700]	Speed: 1563.85 samples/sec	Train-Accuracy_LCS=0.903240
2017-02-04 08:49:52,687 Epoch[0] Batch [9800]	Speed: 1560.48 samples/sec	Train-Accuracy_LCS=0.905164
2017-02-04 08:49:59,106 Epoch[0] Batch [9900]	Speed: 1558.01 samples/sec	Train-Accuracy_LCS=0.903228
2017-02-04 08:50:05,719 Epoch[0] Batch [10000]	Speed: 1512.11 samples/sec	Train-Accuracy_LCS=0.904909

This example contains some training data of 5-20 char long images, ( @30*300 resolution ) , using hyperparams : Hidden=256~512, Layer=2~3, Batch Size = 64~100, and Learning Rate = 0.0005~0.005. 

The original accuracy must be too strict to train the model (if 1 out of 20 char is not correctly predicted,  then the correct part is wasted).

At this morning, this 10-epoch run finally reported a proximately 96.7% accuracy.

2017-02-04 10:46:11,495 Epoch[9] Batch [9600]	Speed: 1544.09 samples/sec	Train-Accuracy_LCS=0.967721
2017-02-04 10:46:17,891 Epoch[9] Batch [9700]	Speed: 1563.57 samples/sec	Train-Accuracy_LCS=0.967243
2017-02-04 10:46:24,317 Epoch[9] Batch [9800]	Speed: 1556.13 samples/sec	Train-Accuracy_LCS=0.968336
2017-02-04 10:46:30,709 Epoch[9] Batch [9900]	Speed: 1564.66 samples/sec	Train-Accuracy_LCS=0.966759
2017-02-04 10:46:37,104 Epoch[9] Batch [10000]	Speed: 1563.60 samples/sec	Train-Accuracy_LCS=0.968629
2017-02-04 10:46:43,535 Epoch[9] Batch [10100]	Speed: 1555.05 samples/sec	Train-Accuracy_LCS=0.967586


Usually a captcha is no longer than 6, a former fixed length run shows that it can achieve 99.1% LCS Accuracy (which means a 94.6% Original Accuracy is ensured).

**Comment by xlvector** [comment]
LGTM, @catalystfrank thanks.

**Comment by fanlu** [comment]
@catalystfrank Can you list your device? I use one gtx1060, can only got about Speed: 306.80 samples/sec.your speed is very very faster

**Comment by catalystfrank** [comment]
CUDA7.5+CuDNNv5.1+gcc4.8.5+OpenCV2.4.9.1+M40(12GB).
@fanlu please move the captcha and image resizing code out of Data Iteration steps. This web I/O latency may be the main bottleneck in your example ( 1060 has almost half single precision TFLOPs of M40).
I've also tried to preload all images into host memory to get faster host-to-device data preparation, however this change doens't help too much , since MXNet has its own data preloading mechanism.

**Comment by fanlu** [comment]
@catalystfrank I first produced the picture and then training, the speed will indeed improve, but many small files, io certainly is a problem, Have you tried im2rec？

**Comment by catalystfrank** [comment]
@fanlu if you can np.float16() your img data and load them into host memory, it might be faster. Another trick is using Solid State Disk(SSD) to store img files. MXNet Im2Rec tool is not used in my approaches.
